client/rest-high-level/src/test/java/org/elasticsearch/client/indexlifecycle/RemoveIndexLifecyclePolicyRequestTests.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.client.indexlifecycle;
+
+import org.elasticsearch.action.support.IndicesOptions;
+import org.elasticsearch.test.ESTestCase;
+
+public class RemoveIndexLifecyclePolicyRequestTests extends ESTestCase {
+
+    private RemoveIndexLifecyclePolicyRequest createTestInstance() {
+        RemoveIndexLifecyclePolicyRequest request = new RemoveIndexLifecyclePolicyRequest(generateRandomStringArray(20, 20, false));
+        if (randomBoolean()) {
+            IndicesOptions indicesOptions = IndicesOptions.fromOptions(randomBoolean(), randomBoolean(), randomBoolean(), randomBoolean(),
+                    randomBoolean(), randomBoolean(), randomBoolean());
+            request.indicesOptions(indicesOptions);
+        }
+        if (randomBoolean()) {
+            request.indices(generateRandomStringArray(20, 20, false));
+        }
+        return request;
+    }
```

Comment:
I think we should implement `mutateInstance` so the hash code and equals tests can test non-equal instances too?

Reply:
The framework changed with the removal the XContent methods.  Added tests for equals and hashcode separately.